test/mapmatch.cc

```diff
@@ -1143,95 +1197,73 @@ TEST(Mapmatch, test_transition_matching) {
-            << "Expected legs with duration " + std::to_string(test_ans_leg_duration[i][k - 1]) +
-                   " but got " + std::to_string(duration);
-      }
-    }
+    auto route_case = R"({"locations)" + test_cases[i].substr(7);
+    auto routed = tester.route(route_case);
+    auto matched = tester.match(test_cases[i]);
+    compare_results(routed, matched);
   }
 }
 
 TEST(Mapmatch, test_intersection_matching) {
   std::vector<std::string> test_cases = {
-      R"({"costing":"auto","format":"osrm","shape_match":"map_snap","shape":[
-          {"lat": 52.0981267, "lon": 5.1296180, "type": "break"},
-          {"lat": 52.0981280, "lon": 5.1297250, "type": "break"}]})",
-      R"({"costing":"auto","format":"osrm","shape_match":"map_snap","shape":[
-          {"lat": 52.0981346, "lon": 5.1300437, "type": "break"},
-          {"lat": 52.0981145, "lon": 5.1309431, "type": "break"},
-          {"lat": 52.0980642, "lon": 5.1314993, "type": "break"},
-          {"lat": 52.0971149, "lon": 5.1311002, "type": "break"}]})",
-      R"({"costing":"auto","format":"osrm","shape_match":"map_snap","shape":[
-          {"lat": 52.0951641, "lon": 5.1285609, "type": "break"},
-          {"lat": 52.0952055, "lon": 5.1292756, "type": "break"},
-          {"lat": 52.0952580, "lon": 5.1301359, "type": "break"},
-          {"lat": 52.0952939, "lon": 5.1309020, "type": "break"},
-          {"lat": 52.0944788, "lon": 5.1304066, "type": "break"}]})"};
-  std::vector<std::pair<int, std::vector<float>>> test_answers = {{1, {7.3}},
-                                                                  {3, {61.7, 41.6, 109.4}},
-                                                                  {4, {49.3, 61, 52.6, 99}}};
+      R"({"shape":[
+          {"lat": 52.0981267, "lon": 5.1296180, "type": "break", "node_snap_tolerance": 0},
+          {"lat": 52.0981280, "lon": 5.1297250, "type": "break", "node_snap_tolerance": 0}],
+          "costing":"auto","shape_match":"map_snap"})",
+      R"({"shape":[
+          {"lat": 52.0981346, "lon": 5.1300437, "type": "break", "node_snap_tolerance": 5},
+          {"lat": 52.0981250, "lon": 5.1309460, "type": "break", "node_snap_tolerance": 5},
+          {"lat": 52.0980642, "lon": 5.1314993, "type": "break", "node_snap_tolerance": 5},
+          {"lat": 52.0980870, "lon": 5.1315040, "type": "break", "node_snap_tolerance": 5}],
+          "costing":"auto","shape_match":"map_snap"})",
+      R"({"shape":[
+          {"lat": 52.0951641, "lon": 5.1285609, "type": "break", "node_snap_tolerance": 5},
+          {"lat": 52.0952055, "lon": 5.1292756, "type": "break", "node_snap_tolerance": 5},
+          {"lat": 52.0952520, "lon": 5.1301350, "type": "break", "node_snap_tolerance": 5},
+          {"lat": 52.0952940, "lon": 5.1309060, "type": "break", "node_snap_tolerance": 5},
+          {"lat": 52.0944788, "lon": 5.1304066, "type": "break", "node_snap_tolerance": 5}],
+          "costing":"auto","shape_match":"map_snap"})"};
```

Comment:
some of these werent actually doing intersection routing so we needed to up the snap tolerance to make it happen